nessie/src/main/java/org/apache/iceberg/nessie/NessieIcebergClient.java

```diff
@@ -323,28 +417,13 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
       throw new AlreadyExistsException("Table already exists: %s", to.name());
     }
 
-    CommitMultipleOperationsBuilder operations =
-        getApi()
-            .commitMultipleOperations()
-            .commitMeta(
-                NessieUtil.buildCommitMetadata(
-                    String.format("Iceberg rename table from '%s' to '%s'", from, to),
-                    catalogOptions))
-            .operation(Operation.Delete.of(NessieUtil.toKey(from)))
-            .operation(Operation.Put.of(NessieUtil.toKey(to), existingFromTable));
-
     try {
-      Tasks.foreach(operations)
-          .retry(5)
-          .stopRetryOn(NessieNotFoundException.class)
-          .throwFailureWhenFinished()
-          .onFailure((o, exception) -> refresh())
-          .run(
-              ops -> {
-                Branch branch = ops.branch((Branch) getRef().getReference()).commit();
-                getRef().updateReference(branch);
-              },
-              BaseNessieClientServerException.class);
+
+      commitRetry(
```

Comment:
I also modified the table operations in this class to use the new `commitRetry` method.

One side-effect of that is that `NessieReferenceConflict` errors are not being retried anymore.

As i voiced somewhere else already, I think conflicts have little to no chance of success when retried, so not only it's useless, but probably bad from a UX perspective, because it only delays the display of the error message to the user.

But I'm happy to put things back as they were if someone disagrees.

Reply:
LGTM :+1:

Reply:
| retrying `NessieReferenceConflict` ... only delays the display of the error message to the user

Agree. Not retrying those is fine IMO.